core/server/kustomization.go

```diff
@@ -14,30 +14,25 @@ import (
 func (cs *coreServer) ListKustomizations(ctx context.Context, msg *pb.ListKustomizationsRequest) (*pb.ListKustomizationsResponse, error) {
 	clustersClient := clustersmngr.ClientFromCtx(ctx)
 
-	clist := clustersmngr.NewClusteredList(func() client.ObjectList {
-		return &kustomizev1.KustomizationList{}
-	})
+	if msg.Namespace != "" {
+		res, err := listKustomizationsInNamespace(ctx, clustersClient, msg.Namespace)
 
-	if err := clustersClient.ClusteredList(ctx, clist, client.InNamespace(msg.Namespace)); err != nil {
-		return nil, err
+		return &pb.ListKustomizationsResponse{
+			Kustomizations: res,
+		}, err
 	}
 
 	var results []*pb.Kustomization
 
-	for n, l := range clist.Lists() {
-		list, ok := l.(*kustomizev1.KustomizationList)
-		if !ok {
+	for _, ns := range cs.cacheContainer.Namespaces() {
+		nsResult, err := listKustomizationsInNamespace(ctx, clustersClient, ns.Name)
+		if err != nil {
+			cs.logger.Error(err, fmt.Sprintf("unable to list kustomizations in namespace: %s", ns.Name))
```

Comment:
Does not fail if it was unable to fetch resources from a given namespace, that way if the user has no access on a namespace, it will fail, but that only means, we don't return resources from that namespace.